pkg/cmd/scylla-manager/server.go

```diff
@@ -230,6 +234,37 @@ func (s *server) makeServers(ctx context.Context) error {
 	return nil
 }
 
+func (s *server) initConfigCacheSvc(ctx context.Context) {
+	s.logger.Info(ctx, "Initializing the clusters configuration cache")
+	defer s.logger.Info(ctx, "Clusters config cache initialized")
+
+	s.configCacheSvc.Init(ctx)
+}
+
+func (s *server) startConfigCacheSvcAsync(ctx context.Context) {
+	go func() {
+		logger := s.logger.Named("Config cache goroutine")
+
+		for {
+			select {
+			case <-ctx.Done():
+				logger.Info(ctx, "Shutdown")
+				return
+			default:
+				func() {
+					defer func() {
+						if r := recover(); r != nil {
+							logger.Error(ctx, "Recovered from panic:", "recover", r)
+						}
+					}()
+
+					s.configCacheSvc.Run(ctx, true)
+				}()
```

Comment:
Yes, makes sense what you wrote that we cannot recover from this panic.
Let me change it the way that we can recover from the panic, by cleaning the map before continuing to loop.

Reply:
Well, if there is a wrong type in the map, resetting the map won't really help as this type will be added to it later anyway. It will most likely create an endless loop of panics and recovers instead of just a single, comprehensible panic.

I'm in favor of dropping this recovery mechanism and letting panics crash SM as we can't truly recover from this state.

Reply:
> Well, if there is a wrong type in the map, resetting the map won't really help as this type will be added to it later anyway. It will most likely create an endless loop of panics and recovers instead of just a single, comprehensible panic.

The problem that I have is that if this service panics then the healtcheck will keep the same config forever. The error will be visible in logs only and just in a single place of the logs.... It's gonna be very easy to overlook it.
Even though I think the service won't panic, because the types are guaranteed by the code, I would prefer to have a possibility of making this (unexpected situation) more visible.

Does it make sense ?